numpy/core/function_base.py

```diff
@@ -72,6 +72,7 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None):
     arange : Similar to `linspace`, but uses a step size (instead of the
              number of samples).
     logspace : Samples uniformly distributed in log space.
+    geomspace : Similar to `logspace`, but with endpoints specified directly.
```

Comment:
I like the idea of the addition, but feel this doesn't help a user who doesn't already know how `logspace` works.  I think it might make more sense to put `geomspace` *first*, as in a way it is in a way the more obvious implementation, and then write `logspace` as "with the end points specified as logarithms" (or something similar).

Reply:
I think you're right.

I copy & pasted the text for `geomspace` from the "See also" from `logspace`. Should we also change the description there as well?

Let's get the wording right, then I'll update the PR.
My suggestions:
```
   ...
    geomspace : Similar to `linspace`, but with numbers spaced evenly on a log scale
                (a geometric progression).
    logspace : Similar to `geomspace`, but with the end points specified as
               logarithms.
```